packages/plugin-core/src/utils/ExtensionUtils.ts

```diff
@@ -439,6 +441,64 @@ export class ExtensionUtils {
     if (previewTheme !== undefined) {
       _.set(trackProps, "previewTheme", previewTheme);
     }
+
+    const allExtensions = vscode.extensions.all;
+    let allNonBuiltInExtensions = allExtensions.filter((extension) => {
+      return !extension.packageJSON.isBuiltin;
+    });
+    if (VSCodeUtils.isDevMode()) {
+      // done to make this work during dev mode
+      allNonBuiltInExtensions = allNonBuiltInExtensions.filter((ext) => {
+        return !ext.extensionPath.includes("packages/plugin-core");
+      });
+    }
+    try {
+      const extensionsDetail = allNonBuiltInExtensions.map((extension) => {
+        const { packageJSON } = extension;
+        const { id, version } = packageJSON;
+        return { id, version };
+      });
+      if (extensionsDetail && extensionsDetail.length > 0) {
+        _.set(trackProps, "extensionsDetail", JSON.stringify(extensionsDetail));
+        _.set(trackProps, "numExtensions", extensionsDetail.length);
+      }
+    } catch (error) {
+      // something went wrong don't track extension detail
+    }
+
+    try {
+      // infer install path from non-built-in extension paths
+      const installPaths = allNonBuiltInExtensions.map((ext) => {
+        // vscode-dot-dir-path/extensions/extension-basename
+        // stripping out last two levels
+        return Utils.dirname(Utils.dirname(URI.file(ext.extensionPath))).fsPath;
+      });
+
+      const pathSet = new Set(installPaths);
+      // make sure every element of install paths are identical.
+      // otherwise the inferred path is not correct
+      let installPath: string;
+      if (pathSet.size === 1) {
+        installPath = installPaths[0];
+      } else {
+        // fail-safe case using current extension context
+        // NOTE: this will not be accurate in dev mode
+        installPath = Utils.dirname(
```

Comment:
why  an't we just use this from the beginning?

Reply:
to avoid false positives in case a custom install path is provided, but after some thought this doesn't really mitigate that at all.

the custom install path case would be very rare so I'll just default to doing this